src/libvirt-common.js

```diff
@@ -135,6 +135,18 @@ export function getElem(xml) {
     return xmlDoc.firstElementChild;
 }
 
+export function parsePoolCapabilities(capsXML) {
```

Comment:
If `virsh` does not support the pool-capabilities command, then capsXML would be empty, right? For some reason it seems to get past this function without crashing (it crashes later, see above), but I figure it couldn't hurt to add an explicit check if `poolCapsElem` is defined.

Reply:
If virsh does not support pool-capabilities this code will not be called, because it's the first parameter of then.

Reply:
AH, right -- the promise would fail there, and `poolCapabilities` would simply be uninitialized. Which reminds me -- please fix the constructor to initialize `poolCapabilities` to `null`. It's nice if the constructor shows a complete picture what `this.state` is.